### PR TITLE
Added DeleteRelatedOrganisationAsync endpoint.

### PR DESCRIPTION
**Pull request by leerogers-bjss**
updated all route references from {id} to {organisationId} in organisationsController to make the swagger/referenced object clearer.
added tests for DeleteRelatedOrganisationAsync endpoint

Story:  [AB#12954](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/12954)
Task: [AB#13820](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13820)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs [hunk 1]
```diff
@@ -261,5 +261,31 @@ public async Task<ActionResult> CreateRelatedOrganisationAsync(Guid id, [FromBod
...
+
+            if (organisation is null)
+            {
+                return NotFound();
+            }
+
+            if (!organisation.RelatedOrganisations.Any(ro => ro.OrganisationId == relatedOrganisationId))
+            {
+                return BadRequest(new ErrorMessageViewModel(FormattableString.Invariant($"The referenced organisation {organisationId} has no relationship to {relatedOrganisationId}.")));
+            }
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs:280]: I think we can omit this. I think a `NoContent()` will be fine in this case – please see additional comments below.

File: src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs [hunk 2]
```diff
@@ -261,5 +261,31 @@ public async Task<ActionResult> CreateRelatedOrganisationAsync(Guid id, [FromBod
...
+
+            var relatedOrganisation = organisation.RelatedOrganisations.Where(ro => ro.OrganisationId == relatedOrganisationId).First();
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs:282]: Semantically, I think `Single()` would fit better. I think we can also simplify by combining the `Where()` and `Single()` to:

```C#
var relatedOrganisation = organisation.RelatedOrganisations.Single(ro => ro.OrganisationId == relatedOrganisationId);
```

File: src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs [hunk 3]
```diff
@@ -261,5 +261,31 @@ public async Task<ActionResult> CreateRelatedOrganisationAsync(Guid id, [FromBod
...
+
+            organisation.RelatedOrganisations.Remove(relatedOrganisation);
+
+            await organisationRepository.UpdateAsync(organisation);
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs:286]: I think we could do something like this here:

```c#
if (organisation.RelatedOrganisations.Remove(relatedOrganisation))
    await organisationRepository.UpdateAsync(organisation);
```
This way we will only update if the related orgs have actually changed and we return `NoContent()` in both cases where we've deleted the relationship and where the relationship didn't exist – I don't think we need to differentiate.

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs [hunk 4]
```diff
@@ -261,5 +261,32 @@ public async Task<ActionResult> CreateRelatedOrganisationAsync(Guid id, [FromBod
...
+
+            if (organisation is null)
+            {
+                return NotFound();
+            }
+
+            if (!organisation.RelatedOrganisations.Any(ro => ro.OrganisationId == relatedOrganisationId))
+            {
+                return NoContent();
+            }
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs:280]: I think we can remove this `if` statement – see comment below.

File: src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs [hunk 5]
```diff
@@ -261,5 +261,32 @@ public async Task<ActionResult> CreateRelatedOrganisationAsync(Guid id, [FromBod
...
+
+            var relatedOrganisation = organisation.RelatedOrganisations.Single(ro => ro.OrganisationId == relatedOrganisationId);
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs:282]: If we update this to `SingleOrDefault`, I think we should be able to remove the `if` statement above without requiring any additional changes (`Remove` will accept a null reference).

**Review by bjss-chris — CHANGES_REQUESTED** [review]
Added a suggestion that I think will make the controller method slightly cleaner. Hopefully you agree 😃 but let me know if not.

File: src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs (hunk 5 above)
Comment (leerogers-bjss) [reply src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs:282]: that's true, but i'm not 100% sure how EF will handle it.
should be fine, but gona double check

**Pull request merged**